### PR TITLE
Only update flow at door when a transition is passed

### LINKED ISSUE #663: The door statistics are wrong
When I run demo 08_big_room. I found the door statistics is wong.

In the inifile, there are 250 pedestrians evacuating from four exits. 

image

But in the door statistics, there are 300 pedestrians evacuating from the four exits.

`2020-01-30 15:17:34.947] [info] Exit ID [0] used by [80] pedestrians. Last passing time [44.58] s [2020-01-30 15:17:34.947] [info] More Information in the file: /home/qiancheng/workspace/jpscore_develop/jpscore/build/bin/results/flow_exit_id_0_big_room_250_traj.txt [2020-01-30 15:17:34.947] [info] Exit ID [1] used by [46] pedestrians. Last passing time [37.46] s [2020-01-30 15:17:34.947] [info] More Information in the file: /home/qiancheng/workspace/jpscore_develop/jpscore/build/bin/results/flow_exit_id_1_big_room_250_traj.txt [2020-01-30 15:17:34.948] [info] Exit ID [2] used by [53] pedestrians. Last passing time [42.54] s [2020-01-30 15:17:34.948] [info] More Information in the file: /home/qiancheng/workspace/jpscore_develop/jpscore/build/bin/results/flow_exit_id_2_big_room_250_traj.txt [2020-01-30 15:17:34.948] [info] Exit ID [3] used by [121] pedestrians. Last passing time [46.21] s [2020-01-30 15:17:34.948] [info] More Information in the file: /home/qiancheng/workspace/jpscore_develop/jpscore/build/bin/results/flow_exit_id_3_big_room_250_traj.txt`

### PR DESCRIPTION
**Pull request by schroedtert**
The usage of the door is now only incremented if the pedestrian has changed its room. 
Fixes #663.

### CODE REVIEW

File: libcore/src/pedestrian/Pedestrian.cpp [hunk 1]
```diff
@@ -1160,6 +1158,9 @@ bool Pedestrian::Relocate(std::function<void(const Pedestrian &)> flowupdater)
 #pragma omp critical(SetEgressTime)
                 allRooms.at(oldRoomID)->SetEgressTime(
                     GetGlobalTime()); //set Egresstime to old room //@todo: ar.graf : GetRoomID() yields NEW room
+#pragma omp critical(flowupdater)
+                flowupdater(
+                    *this); //@todo: ar.graf : this call should move into a critical region? check plz
```
Comment (chraibi) [inline libcore/src/pedestrian/Pedestrian.cpp]: you don't want to remove this todo?

File: systemtest/statistics_tests/door_statistics/01_door_statistics.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+from utils import SUCCESS, FAILURE
+from JPSRunTest import JPSRunTestDriver
+
+import xml.etree.cElementTree as ET
+
+
+def readFlow(numPeds):
+    filename = 'flow_exit_id_0_traj_group-number_{:d}.txt'.format(int(numPeds))
+    flowfile = os.path.join('results', filename)
+    logging.error('try to read: {}'.format(flowfile))
```
Comment (chraibi) [inline systemtest/statistics_tests/door_statistics/01_door_statistics.py]: `error` --> `info`

File: systemtest/statistics_tests/door_statistics/01_door_statistics.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python3
+#################################################################################
+# Check if the number of pedestrians using the exit is the same as the number of
+# pedestrians in the simulation. Hence checking if the door statistics are correct
+#################################################################################
+import os
+from sys import argv, path
+import logging
+import pandas as pd
+import matplotlib.pyplot as plt
```
Comment (chraibi) [inline systemtest/statistics_tests/door_statistics/01_door_statistics.py]: matplotlib not needed

File: libcore/src/pedestrian/Pedestrian.cpp (hunk 1 above)
Comment (zeroset) [reply libcore/src/pedestrian/Pedestrian.cpp]: The format looks pretty broken?

Comment (zeroset) [reply libcore/src/pedestrian/Pedestrian.cpp]: Ah no, the comment should be behind ... anyways, ugly :D

File: systemtest/statistics_tests/door_statistics/01_door_statistics.py (hunk 3 above)
Comment (schroedtert) [reply systemtest/statistics_tests/door_statistics/01_door_statistics.py]: removed

File: systemtest/statistics_tests/door_statistics/01_door_statistics.py (hunk 2 above)
Comment (schroedtert) [reply systemtest/statistics_tests/door_statistics/01_door_statistics.py]: changed

File: libcore/src/pedestrian/Pedestrian.cpp (hunk 1 above)
Comment (schroedtert) [reply libcore/src/pedestrian/Pedestrian.cpp]: removed, and also changed the position of the other comment.

**Comment by anna-braun** [comment]
I tested the PR, but the statistics are still wrong. I tested it with exit strategy 8 and 9. For 9 the statistic of one door shows more than 270000 agents where only 1450 should be.

**Comment by anna-braun** [comment]
> I tested the PR, but the statistics are still wrong. I tested it with exit strategy 8 and 9. For 9 the statistic of one door shows more than 270000 agents where only 1450 should be.

And the door status "temp_close" is overwritten by the door flow again.

**Comment by schroedtert** [comment]
Will be opened again, if problem is really solved. Seems there is missing something.

**Pull request closed without merging**